components/encoder/single/single_encoder_test.go

```diff
@@ -231,22 +223,46 @@ func TestEncoder(t *testing.T) {
 	})
 }
 
-func MakeBoard(t *testing.T) *fakeboard.Board {
-	interrupt, _ := fakeboard.NewDigitalInterruptWrapper(board.DigitalInterruptConfig{
-		Name: "10",
-		Pin:  "10",
-	})
+func MakeBoard(t *testing.T, name, pinname string) board.Board {
+
+	b := inject.NewBoard(name)
+	i := inject.DigitalInterrupt{}
+	i.NameFunc = func() string {
+		return testPinName
+	}
+	i.TickFunc = func(ctx context.Context, high bool, nanoseconds uint64) error {
+		time.Sleep(50 * time.Microsecond)
+		return nil
+
+	}
+	i.RemoveCallbackFunc = func(c chan board.Tick) {}
 
-	interrupts := map[string]*fakeboard.DigitalInterruptWrapper{
-		"10": interrupt,
+	b.DigitalInterruptByNameFunc = func(name string) (board.DigitalInterrupt, bool) {
+		return &i, true
 	}
+	b.StreamTicksFunc = func(ctx context.Context, interrupts []board.DigitalInterrupt, ch chan board.Tick, extra map[string]interface{}) error {
```

Comment:
note this is in no way thread safe or complete, but basically a stupid flow for streaming ticks.